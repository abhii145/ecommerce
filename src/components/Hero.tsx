import Carousel from "./Carousel"
import banner1 from "../assets/banner1.svg"

const Hero = () => {
  const slides = [
    {
      type: "gradient",
      background: "bg-gradient-to-r from-blue-500 to-indigo-600",
      title: "Welcome to ShopEasy",
      description: "Discover amazing products at unbeatable prices.",
      buttonText: "Shop Now",
      link: "/products",
    },
    {
      type: "image",
      imageSrc: banner1,
      altText: "Banner showcasing new arrivals",
      link: "/products",
    },
    {
      type: "image",
      imageSrc: banner1,
      altText: "Banner showcasing new arrivals",
      link: "/new-arrivals",
    },
  ]

  return (
    <section aria-labelledby="hero-heading">
      <h1 id="hero-heading" className="sr-only">Welcome to ShopEasy</h1>
      <Carousel slides={slides} />
    </section>
  )
}

export default Hero
