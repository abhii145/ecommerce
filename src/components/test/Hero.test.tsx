import React from "react"
import { render } from "@testing-library/react"
import { BrowserRouter as Router } from "react-router"
import Hero from "../Hero"

describe("Hero Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Router>
        <Hero />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Hero component", () => {
    const { getAllByText } = render(
      <Router>
        <Hero />
      </Router>
    )
    expect(getAllByText(/welcome to shopeasy/i)[0]).toBeInTheDocument()
    expect(getAllByText(/shop now/i)[0]).toBeInTheDocument()
  })
})
