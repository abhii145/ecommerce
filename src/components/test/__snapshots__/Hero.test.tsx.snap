// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="hero-heading"
  >
    <h1
      class="sr-only"
      id="hero-heading"
    >
      Welcome to ShopEasy
    </h1>
    <div
      aria-roledescription="carousel"
      class="relative"
    >
      <div
        aria-label="Slide 1 of 3"
        aria-roledescription="slide"
        class="relative bg-gradient-to-r from-blue-500 to-indigo-600 text-white"
        role="group"
        style="height: 500px; opacity: 0; transform: scale(0.9);"
      >
        <div
          class="container mx-auto text-center h-full flex flex-col justify-center items-center"
        >
          <h1
            class="text-5xl font-bold"
          >
            Welcome to ShopEasy
          </h1>
          <p
            class="mt-4 text-lg max-w-md mx-auto"
          >
            Discover amazing products at unbeatable prices.
          </p>
          <button
            class="mt-6 px-6 py-2 bg-white text-blue-600 rounded-lg shadow hover:bg-blue-100"
          >
            <a
              data-discover="true"
              href="/products"
            >
              Shop Now
            </a>
          </button>
        </div>
      </div>
      <div
        class="absolute top-1/2 left-0 transform -translate-y-1/2 px-4"
      >
        <button
          aria-label="Previous Slide"
          class="text-white text-2xl bg-black bg-opacity-50 p-2 rounded-full"
        >
          ‹
        </button>
      </div>
      <div
        class="absolute top-1/2 right-0 transform -translate-y-1/2 px-4"
      >
        <button
          aria-label="Next Slide"
          class="text-white text-2xl bg-black bg-opacity-50 p-2 rounded-full"
        >
          ›
        </button>
      </div>
      <div
        class="absolute bottom-4 left-1/2 transform -translate-x-1/2 flex space-x-2"
      >
        <button
          aria-label="Go to slide 1"
          class="w-3 h-3 rounded-full bg-green-500"
        />
        <button
          aria-label="Go to slide 2"
          class="w-3 h-3 rounded-full bg-gray-600"
        />
        <button
          aria-label="Go to slide 3"
          class="w-3 h-3 rounded-full bg-gray-600"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
